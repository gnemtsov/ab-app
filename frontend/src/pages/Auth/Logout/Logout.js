import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import * as actionTypes from "../../../store/actionTypes";

class Logout extends Component {
    componentDidMount() {
        this.props.onLogout();
    }

    render() {
        return <Redirect to="/" />;
    }
}

const mapDispatchToProps = dispatch => {
    return {
        onLogout: () => dispatch({
            type: actionTypes.S_LOGOUT
        })
    };
};

export default connect(null, mapDispatchToProps)(Logout);