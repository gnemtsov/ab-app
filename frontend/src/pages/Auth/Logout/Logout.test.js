import React from 'react';

import configureStore from 'redux-mock-store';

import { configure, render, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Logout from './Logout';

configure({ adapter: new Adapter() });

describe('Logout', () => {
	it('Renders correctly', () => {
		const mockStore = configureStore();
		const store = mockStore({}); 
		
		const wrapper = shallow(<Logout store={store}/>);
		expect(wrapper).toMatchSnapshot();
	});
	
	/*it('Dispatches logout', () => {
		const mockStore = configureStore();
		const store = mockStore({}); 
		
		const wrapper = shallow(<Logout store={store}/>);
		return Promise.resolve(wrapper)
			.then(
				() => {
					wrapper.update();
					console.log(store.getActions());
				}
			);	
	});*/
})
