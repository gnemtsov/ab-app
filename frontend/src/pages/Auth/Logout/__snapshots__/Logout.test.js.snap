// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Renders correctly 1`] = `
<Logout
  onLogout={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
